composition.js

```diff
@@ -0,0 +1,199 @@
+"use strict"
+
+class Ingredient {
+    constructor(options) {
+        this.name = options['name']
+        this.amount = options['amount']
+        this.unit = options['unit']
+    }
+}
+
+class Cookie {
+    constructor(ingredients) {
+        this.name = null
+        this.status = 'mentah'
+        this.ingredients = []
+        this.has_sugar = null
+        this.has_peanut = null
+        //ingredients is still an array of objects. to instantiate several new Ingredient object
+        ingredients.forEach(x => {
+            //use composition of Ingredient class
+            this.ingredients.push(new Ingredient(x))
+            //check if there is any sugar/peanut
+            if(x.name === 'sugar') {
+                this.has_sugar = true
+            }
+            else if(x.name.indexOf('peanut') >= 0) {
+                this.has_peanut = true
+            }
+        })
+    }
+
+    bake() {
+        this.status = 'selesai dimasak'
+    }
+
+}
+
+class PeanutButter extends Cookie {
+    constructor(ingredientObj) {
+        super(ingredientObj)
```

Comment:
```suggestion
        super('peanut butter', ingredientObj)
```
karena setiap cookie akan punya nama, lebih baik dimasukkan ke parameter di constructor parentnya (class Cookie)